frontend/src/pages/user-list.tsx

```diff
@@ -1,31 +1,74 @@
 import useSWR from 'swr'
-import React from 'react'
 import Link from 'next/link'
+import { toast } from 'react-toastify'
+import React, { useState } from 'react'
 import { MdVerified } from 'react-icons/md'
 
+import axios from '~/lib/axios'
+import { useAuth } from '~/hooks/auth'
 import { fetcher } from '~/lib/fetcher'
 import Avatar from '~/components/Avatar'
+import { IUser } from '~/data/interfaces'
 import { Spinner } from '~/utils/Spinner'
 import Layout from '~/layouts/userLayout'
-import { IUser } from '~/data/interfaces'
 import { classNames } from '~/utils/classNames'
 
 const UserList = (): JSX.Element => {
-  const { data: users } = useSWR('/api/user-privilege', async () => fetcher('/api/user-privilege'), {
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const { user: author } = useAuth({
+    middleware: 'auth'
+  })
+  
+  const { data: users, mutate } = useSWR('/api/user-privilege', async () => fetcher('/api/user-privilege'), {
     refreshInterval: 1000,
     revalidateOnMount: true
   })
 
+  const handleFollow = async (user: IUser): Promise<void> => {
+    setLoading(true)
+    const current_id = user?.followers?.map(({ id }) => id)
+
+    if (current_id == author?.id) {
+      // UNFOLLOW
+      return await 
+              axios
+                .patch(`/api/follow/${user?.id}`, { following_id: user?.id })
+                .then(async () =>{
+                  await mutate()
+                  toast.success(`You unfollow ${user?.name}`)
+                })
+                .finally(() => setLoading(false))
+    } else {
+      // FOLLOW
+      return await 
+              axios
+                .post('/api/follow', {
+                  following_id: user?.id
+                })
+                .then(async () => {
+                  await mutate()
+                  toast.success(`You follow ${user?.name}`)
+                })
+                .finally(() => setLoading(false))
+    }
+  }
+
+  const userFollowStatus = (user: IUser): string => {
+    const current_id = user?.followers?.map(({ id }) => id)
+    return current_id == author?.id ? 'Unfollow' : 'Follow'
+  }
+
```

Comment:
since this is a repeating function, much better to create a separate helper file and just call that function